Userland/Libraries/LibGUI/ItemListModel.h

```diff
@@ -77,6 +77,26 @@ class ItemListModel : public Model {
         return {};
     }
 
+    virtual bool is_searchable() const override { return !IsTwoDimensional; }
+    virtual Vector<GUI::ModelIndex, 1> matches(StringView const& searching, unsigned flags, GUI::ModelIndex const&) override
+    {
+        if constexpr (IsTwoDimensional)
+            return {};
```

Comment:
Actually 2d model search *is* possible, but I didn’t put it there, because the code and the result just wasn’t looking good imo.

```cc
if constexpr (IsTwoDimensional) {
    for (auto it = m_data.begin(); it != m_data.end(); ++it) {
        for (auto it2d = (*it).begin(); it2d != (*it).end(); ++it2d) {
            GUI::ModelIndex index = this->index(it.index(), it2d.index());
            // ...
```

![2d](https://user-images.githubusercontent.com/16520278/130521523-e0466e4b-00b3-4910-a81c-35c45f11e151.png)

Also, 2D list models are used only in the CSV import/export window.

Reply:
Now I think it doesn’t look that bad, or maybe my eyes just got used to it.
Anyway, I’ll update this PR in a moment and add 2d search support.